### PR TITLE
Task/boko 5 add policy methods

### PR DESCRIPTION
**Pull request by Dmitry-VV**

### CODE REVIEW

File: storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/policy/builder/StatementBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.icerockdev.service.storage.s3.policy.builder
+
+import com.icerockdev.service.storage.s3.policy.dto.ActionEnum
+import com.icerockdev.service.storage.s3.policy.dto.EffectEnum
+import com.icerockdev.service.storage.s3.policy.dto.Principal
+import com.icerockdev.service.storage.s3.policy.dto.Statement
+import java.lang.Exception
+
+class StatementBuilder {
+    var effect: EffectEnum? = null
```
Comment (oyakovlev) [inline storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/policy/builder/StatementBuilder.kt:10]: effect not nullable in Statement, why is nullable in this class?

File: storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/S3StorageImpl.kt [hunk 2]
```diff
@@ -215,6 +240,62 @@ class S3StorageImpl(private val client: S3Client, private val preSigner: S3Presi
...
+                logger.error(e.localizedMessage, e)
+            }
+            null
+        }
+    }
+
+    override fun putBucketPolicy(bucket: String, configure: PolicyBuilder.() -> Unit, confirmRemoveSelfBucketAccess: Boolean): Boolean {
+        return try {
+            val p = PolicyBuilder().apply(configure).build()
+            println("Size: " + p.toByteArray().size)
```
Comment (oyakovlev) [inline storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/S3StorageImpl.kt:261]: this is superfluous here

File: storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/IS3Storage.kt [hunk 3]
```diff
@@ -49,6 +54,16 @@ interface IS3Storage {
     fun generateFileKey(): String {
         return UUID.randomUUID().toString().replace("-","/")
     }
+
+    fun getBucketPolicy(bucket: String): String?
+
+    fun putBucketPolicy(bucket: String, configure: PolicyBuilder.() -> Unit, confirmRemoveSelfBucketAccess: Boolean = false): Boolean
```
Comment (oyakovlev) [inline storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/IS3Storage.kt:60]: transfer the function `configure`  as the last argument

File: storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/S3StorageImpl.kt (hunk 2 above)
Comment (Dmitry-VV) [reply storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/S3StorageImpl.kt:261]: Fix

File: storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/IS3Storage.kt (hunk 3 above)
Comment (Dmitry-VV) [reply storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/IS3Storage.kt:60]: Done

File: storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/policy/builder/StatementBuilder.kt (hunk 1 above)
Comment (Dmitry-VV) [reply storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/policy/builder/StatementBuilder.kt:10]: To check the builder data

Comment (oyakovlev) [reply storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/policy/builder/StatementBuilder.kt:10]: need check it in `build()` dunction

Comment (Dmitry-VV) [reply storage-service/src/main/kotlin/com/icerockdev/service/storage/s3/policy/builder/StatementBuilder.kt:10]: Done

**Review by oyakovlev — APPROVED** [review]

**Pull request merged**